src/dotnet/Fable.Compiler/Transforms/FSharp2Fable.fs

```diff
@@ -804,9 +804,8 @@ let private transformDeclarations (com: FableCompiler) fsDecls =
                         EntityName = entityName
                         IsPublic = isPublicEntity ent }
                     [Fable.UnionConstructor info |> Fable.ConstructorDeclaration]
-                | None when ent.IsFSharpRecord
-                        || ent.IsFSharpExceptionDeclaration
-                        || ((ent.IsClass || ent.IsValueType) && not ent.IsMeasure && not(hasImplicitConstructor ent)) ->
+                | None when (ent.IsFSharpRecord || ent.IsValueType || ent.IsFSharpExceptionDeclaration)
+                        || (ent.IsClass && not ent.IsMeasure && not (hasImplicitConstructor ent)) ->
```

Comment:
Is this necessary? Value types can also have implicit constructors [as in here](https://github.com/fable-compiler/Fable/blob/91cf4ea0b5cf7e0ae7a2dc4bad5e751db18cee85/tests/Main/TypeTests.fs#L181) and if we remove this check, there will be duplicated code (the auto-generated constructor and the user-defined constructors, besides the `inherits` instructions).

Reply:
I think we need it, cause value type records and value types compile to the same AST, the only difference seems to be the constructor is marked as implicit, and it's body is a bit different, but that's beside the point. They both still need to inherit from Types.Record.
If there is any differentiation needed it can happen on the generation side (in FableToBabel.fs), but I think its fine the way it is.

Reply:
Ok, thanks for the clarification. I think I'll revert this change and add the inheritance in Fable2Babel when transforming the implicit constructor for value types, as it was a couple of commits ago.